app/src/main/java/com/hedvig/app/feature/home/ui/HomeAdapter.kt

```diff
@@ -230,6 +233,26 @@ class HomeAdapter(
             }
         }
 
+        class UpcomingRenewal(parent: ViewGroup) :
+            ViewHolder(parent.inflate(R.layout.renewal_list)) {
+            private val binding by viewBinding(RenewalListBinding::bind)
+            override fun bind(
+                data: HomeModel,
+                fragmentManager: FragmentManager,
+                retry: () -> Unit,
+                requestBuilder: RequestBuilder<PictureDrawable>,
+                tracker: HomeTracker,
+                marketProvider: MarketProvider
+            ): Any? = with(binding) {
+                if (data !is HomeModel.UpcomingRenewal) {
+                    return invalid(data)
+                }
+                root.adapter = UpcomingRenewalAdapter(tracker).also {
```

Comment:
We should not have a nested `RecyclerView` here. Instead, directly generate one card to be added to this `RecyclerView`.

Reply:
What is the gain by doing it that way?

Reply:
One `RecyclerView` will mean that we actually recycle properly.